deploy/hub.py

```diff
@@ -388,28 +282,37 @@ def deploy(self, auth_provider, proxy_secret_key, skip_hub_health_test=False):
                 'helm', 'upgrade', '--install', '--create-namespace', '--wait',
                 '--namespace', self.spec['name'],
                 self.spec['name'], os.path.join('hub-templates', self.spec['template']),
-                # Ordering matters here - config explicitly mentioned in `hubs.yaml` should take
+                # Ordering matters here - config explicitly mentioned in clu should take
                 # priority over our generated values. Based on how helm does overrides, this means
-                # we should put the config from hubs.yaml last.
+                # we should put the config from config/hubs last.
                 '-f', generated_values_file.name,
                 '-f', values_file.name,
             ]
 
             print(f"Running {' '.join(cmd)}")
+            # Can't test without deploying, since our service token isn't set by default
             subprocess.check_call(cmd)
 
             if not skip_hub_health_test:
 
+                # FIXMEL: Clean this up
                 if self.spec['template'] != 'base-hub':
                     service_api_token = generated_values["base-hub"]["jupyterhub"]["hub"]["services"]["hub-health"]["apiToken"]
                 else:
                     service_api_token = generated_values["jupyterhub"]["hub"]["services"]["hub-health"]["apiToken"]
 
+                domain = self.spec["domain"]
+                if type(domain) == str:
+                    hub_url = f'https://{domain}'
+
+                else:
+                    hub_url = f'https://{random.choice(domain)}'
```

Comment:
Should not be using always one (the first? the last?) instead of a random one? Just curious about why are you adding that layer of variability...

Reply:
Nothing particular, I was just porting the old logic. We should probably instead strive to just have one domain per hub though.